### PR TITLE
Add files

### PR DESCRIPTION
**Pull request by hewison-chris**
This moves the binary serialization implementation from the Agora module into this submodule library.

### CODE REVIEW

**Review by linked0 — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+on: [ push, pull_request ]
+
+jobs:
+  main:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-18.04, macOS-10.15, windows-2019 ]
+        dc: [ ldc-master, ldc-latest, ldc-1.20.0, dmd-latest, dmd-2.090.1 ]
```
Comment (kchulj) [inline .github/workflows/main.yml]: > `ldc-1.20.0`

Is it for workaround?

Comment (Geod24) [reply .github/workflows/main.yml]: It's just what `crypto` is doing. But crypto doesn't rely on `-preview=in`, and this does, so the `toolChainRequirements` should reflect this

**Review by kchulj — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (hewison-chris) [reply .github/workflows/main.yml]: I copied it from crypto submodule workflow. 
@Geod24 can we use ldc-1.25.0 here?

Comment (kchulj) [reply .github/workflows/main.yml]: Perhaps `crypto` should use `ldc-1.25.0` too.

**Pull request merged**